### PR TITLE
Fix the changed GQA behavior by #837.

### PR DESCRIPTION
**Pull request by ds-hwang**
When num_head=4, num_kv_head=2, and seq_len=5,
```
k = 0 1 2 3 4
    5 6 7 8 9
```

the original impl convert it to
  `jnp.repeat(key, 2, axis=head_axis)`
```
k = 0 1 2 3 4
    0 1 2 3 4
    5 6 7 8 9
    5 6 7 8 9
```
, not
```
k = 0 1 2 3 4
    5 6 7 8 9
    0 1 2 3 4
    5 6 7 8 9
```

This PR maintains the original behavior.

### CODE REVIEW

**Comment by ds-hwang** [comment]
@ruomingp Could you review?

**Review by markblee — APPROVED** [review]
